lib/pure/httpcore.nim

```diff
@@ -183,6 +183,13 @@ proc `[]=`*(headers: HttpHeaders, key, value: string) =
   ## Replaces any existing values.
   headers.table[headers.toCaseInsensitive(key)] = @[value]
 
+iterator iter*(headers: HttpHeaders, key: string): string =
+  ## Yields each value associated with the requested key
+  ## If no such key exists, this does nothing.
+  if headers.table.contains(headers.toCaseInsensitive(key)):
+    for value in headers.table[headers.toCaseInsensitive(key)]:
+      yield value
+
```

Comment:
We would need to come up with a good naming scheme for this kind of iterators. I would prefer to just remove it and for you to inline this code above.

Reply:
I agree that the naming is poor, but I think this style of accessor is very valuable because it offers exception-free access, similar to range-based for loops on lists. Maybe simply calling it `range` instead of `iter`? 

I don't mind inlining it for now, but it's actually a little awkward because `toCaseInsensitive` isn't exported, so I'd have to make it public now. For now, I haven't made this change, but I will do so if we're ok with exporting that property (or renaming the iterator).

Reply:
You don't need the `toCaseInsensitive` call though. `[]` and `contains` already handle this.

Reply:
Fixed.